### ISSUE TITLE
nginx-ingress - creates 0.0.0.0/0 exposed loadbalancer on AWS

### ISSUE THREAD

**Issue by kartik-moolya** [helm/charts#22366]
**Ingress controller creates an ELB which is EXPOSED TO THE WORLD**

**Version of Helm and Kubernetes**: Helm2 and Kubernetes 1.14

**Which chart**: nginx-ingress

**What happened**: when applied this chart creates an ELB which is exposed to the world (security group attached exposes 0.0.0.0/0)

**What you expected to happen**: We should be able to configure desired inbounds for the load balancer, there should be an extra configurable variable which would allow

**How to reproduce it** (as minimally and precisely as possible): just try deploying it on AWS

**Anything else we need to know**:

I did edit the security group later manually, but the controller seems to be adding those entries back

**Values.yaml** 
```controller:
  priorityClassName: non-system-high-priority
  tolerations:
  - key: "node"
    operator: "Equal"
    value: "compute"
    effect: "NoSchedule"
  kind: "DaemonSet"
  daemonset:
    useHostPort: "true"
    hostPorts:
      http: '80'
      https: '443'
  config:
    server-name-hash-max-size: '1024'
    server-name-hash-bucket-size: '128'
  podAnnotations:
    prometheus.io/scrape: "true"
    prometheus.io/port: "10254"
  metrics:
    enabled: "true"
  service:
    annotations: 
      service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:eu-west-1:XXXXXXXXX:certificate/XXXXXXXXXXXX
      service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
      service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
      service.beta.kubernetes.io/aws-load-balancer-ssl-ports: https
      service.beta.kubernetes.io/aws-load-balancer-connection-idle-timeout: '3600'
      service.beta.kubernetes.io/aws-load-balancer-internal: "true"
    nodePorts:
      http: "32080"
      https: "32443"
    type: LoadBalancer
    targetPorts:
      https: http
defaultBackend:
  priorityClassName: non-system-high-priority
```

**Comment by tuananhnguyen-ct** [comment]
Have you tried using `service.beta.kubernetes.io/aws-load-balancer-security-groups` or `service.beta.kubernetes.io/aws-load-balancer-extra-security-groups` annotation?

**Comment by kartik-tavisca** [comment]
Thanks for the response,
nope, I havent tried this annotation
So if i use this annotation will it override the default inbound rules ? or just add to it ?
i mean, the default inbound rules allow https to the world which I do not want

> Have you tried using `service.beta.kubernetes.io/aws-load-balancer-security-groups` or `service.beta.kubernetes.io/aws-load-balancer-extra-security-groups` annotation?

**Comment by kartik-tavisca** [comment]
> Have you tried using `service.beta.kubernetes.io/aws-load-balancer-security-groups` or `service.beta.kubernetes.io/aws-load-balancer-extra-security-groups` annotation?

Thanks, I think this is what I was looking for. will try this

**Comment by kartik-tavisca** [comment]
There's an issue when I do it though, the controller fails to create the ELB, I see below when I describe the service

`kubectl describe service nginx-ingress-controller`

>Error creating load balancer (will retry): failed to ensure load balancer for service nginx-ingress/nginx-ingress-controller: error revoking security group ingress: "InvalidPermission.NotFound: The specified rule does not exist in this security group.\n\tstatus code: 400, request id: e29bad0a-0f8e-48fd-80f5-505a54e4a11c"

I tried attaching even the `Administrator Access` to the worker node temporarily but I still get this error

I just added this in the above annotations
```
service.beta.kubernetes.io/aws-load-balancer-security-groups: sg-0XXXXXXXXXX```

**Comment by kartik-tavisca** [comment]
So I was able to add a security group by adding below annotation

`service.beta.kubernetes.io/aws-load-balancer-security-groups: sg-0f995bXXXXXXX`

but it did not do any good.
my security group just allowed 2 inbound rules, but when nginx-ingress mapped this security group with the ELB it overwrote the security group inbound rules with its default inbound rules
Below are the screenshots:

**security group inbound rules before nginx helm chart was installed** 

![image](https://user-images.githubusercontent.com/56674362/81840387-8e033880-9566-11ea-8d51-8dd8b7618e73.png)


**security group inbound rules after nginx helm chart was installed** :cry: 

![image](https://user-images.githubusercontent.com/56674362/81840807-1bdf2380-9567-11ea-839e-ec07b64699c4.png)

@tuananhnguyen-ct

**Comment by tuananhnguyen-ct** [comment]
It's not an issue with nginx-ingress chart, you might want to check this https://github.com/kubernetes/cloud-provider-aws/issues/65 and similar issues there.

**Comment by kartik-tavisca** [comment]
> It's not an issue with nginx-ingress chart, you might want to check this [kubernetes/cloud-provider-aws#65](https://github.com/kubernetes/cloud-provider-aws/issues/65) and similar issues there.

so as per this below update in that issue, we will not have a fix until we push k8s to 1.18 ?

https://github.com/kubernetes/cloud-provider-aws/issues/65#issuecomment-591928611
